docs/src/pages/components/data-grid/editing/ValidateServerNameGrid.tsx

```diff
@@ -0,0 +1,114 @@
+import { makeStyles, Theme } from '@material-ui/core/styles';
+import * as React from 'react';
+import {
+  GridColumns,
+  GridEditCellPropsParams,
+  GridRowsProp,
+  useGridApiRef,
+  getThemePaletteMode,
+  XGrid,
+} from '@material-ui/x-grid';
+
+const columns: GridColumns = [
+  { field: 'name', headerName: 'MUI Contributor', width: 180, editable: true },
+];
+
+const rows: GridRowsProp = [
+  {
+    id: 1,
+    name: 'Damien',
+  },
+  {
+    id: 2,
+    name: 'Olivier',
+  },
+  {
+    id: 3,
+    name: 'Danail',
+  },
+  {
+    id: 4,
+    name: 'Matheus',
+  },
+  {
+    id: 5,
+    name: 'You?',
+  },
+];
+const useStyles = makeStyles((theme: Theme) => {
+  const backgroundColor =
+    getThemePaletteMode(theme.palette) === 'dark' ? '#376331' : 'rgb(217 243 190)';
+  return {
+    root: {
+      '& .MuiDataGrid-cellEditable': {
+        backgroundColor,
+      },
+      '& .MuiDataGrid-cellEditing': {
+        backgroundColor: 'rgb(255,215,115, 0.19)',
+        color: '#1a3e72',
+      },
+      '& .Mui-error': {
+        backgroundColor: 'rgb(126,10,15, 0.1)',
+        color: '#750f0f',
+      },
+    },
+  };
+});
+
+let promiseTimeout: any;
+function validateName(username: string): Promise<boolean> {
+  const existingUsers = rows.map((row) => row.name.toLowerCase());
+
+  return new Promise<any>((resolve) => {
+    promiseTimeout = setTimeout(() => {
+      resolve(existingUsers.indexOf(username.toLowerCase()) === -1);
+    }, Math.random() * 500 + 100);
```

Comment:
Maybe it would make sense to keep it as it might not be obvious why

```suggestion
    }, Math.random() * 500 + 100); // simulate network latency
```